src/ir/new_ir_test.cc

```diff
@@ -0,0 +1,63 @@
+#include "src/common/testing/gtest.h"
+#include "src/ir/context.h"
+#include "src/ir/operation.h"
+#include "src/ir/types/primitive_type.h"
+#include "src/ir/value.h"
+
+namespace raksha::ir {
+
+TEST(NewIrTest, TrySomeStuff) {
+  Context context;
+
+  // particle P1
+  //  .bar: reads Bar {}
+  //   foo: writes Foo {}
+  //   claim foo.a is userSelection
+  //   claim foo.x is derived from bar.y
+  //
+
+  // Using PrimitiveType for illustrative purposes.
+  // In reality we will have to create the correct types.
+  // For generic operators, we should also create a TypeVariable type.
+
+  const Operator* select_op = context.RegisterOperator(
+      OperatorBuilder("sql.select")
+          .AddInput("input", std::make_unique<types::PrimitiveType>())
+          .AddOutput("output", std::make_unique<types::PrimitiveType>())
```

Comment:
That crossed my mind as well. I initially only had `types::Type*` in the operator with the assumption that there will be a central allocator. However, we will still need to have a light-weight wrapper of `Type` that the operator should own. It will be needed for type checking and type inference.